Makefile

```diff
@@ -26,85 +26,94 @@
 #
 
 # This repo's root import path (under GOPATH).
-ROOT := github.com/caicloud/golang-template-project
+ROOT             := github.com/caicloud/golang-template-project
 
 # Target binaries. You can build multiple binaries for a single project.
-TARGETS := admin controller
+TARGETS          := $(patsubst cmd/%,%,$(wildcard cmd/*))
 
 # Container image prefix and suffix added to targets.
 # The final built images are:
 #   $[REGISTRY]/$[IMAGE_PREFIX]$[TARGET]$[IMAGE_SUFFIX]:$[VERSION]
 # $[REGISTRY] is an item from $[REGISTRIES], $[TARGET] is an item from $[TARGETS].
-IMAGE_PREFIX ?= $(strip template-)
-IMAGE_SUFFIX ?= $(strip )
+IMAGE_PREFIX     ?= $(strip template-)
+IMAGE_SUFFIX     ?= $(strip )
 
 # Container registries.
-REGISTRY ?= cargo.dev.caicloud.xyz/release
+REGISTRY         ?= cargo.dev.caicloud.xyz/release
 
 # Container registry for base images.
-BASE_REGISTRY ?= cargo.caicloud.xyz/library
+BASE_REGISTRY    ?= cargo.caicloud.xyz/library
 
 #
 # These variables should not need tweaking.
 #
 
 # It's necessary to set this because some environments don't link sh -> bash.
-export SHELL := /bin/bash
+export SHELL     := /bin/bash
 
 # It's necessary to set the errexit flags for the bash shell.
 export SHELLOPTS := errexit
 
 # Project main package location (can be multiple ones).
-CMD_DIR := ./cmd
+CMD_DIR          := ./cmd
 
 # Project output directory.
-OUTPUT_DIR := ./bin
+OUTPUT_DIR       := ./bin
 
 # Build direcotory.
-BUILD_DIR := ./build
+BUILD_DIR        := ./build
 
 # Current version of the project.
-VERSION ?= $(shell git describe --tags --always --dirty)
+VERSION          ?= $(shell git describe --tags --always --dirty)
 
 # Available cpus for compiling, please refer to https://github.com/caicloud/engineering/issues/8186#issuecomment-518656946 for more information.
-CPUS ?= $(shell /bin/bash hack/read_cpus_available.sh)
+CPUS             ?= $(shell /bin/bash hack/read_cpus_available.sh)
 
 # Track code version with Docker Label.
-DOCKER_LABELS ?= git-describe="$(shell date -u +v%Y%m%d)-$(shell git describe --tags --always --dirty)"
+DOCKER_LABELS    ?= git-describe="$(shell date -u +v%Y%m%d)-$(shell git describe --tags --always --dirty)"
 
 # Golang standard bin directory.
-GOPATH ?= $(shell go env GOPATH)
-BIN_DIR := $(GOPATH)/bin
-GOLANGCI_LINT := $(BIN_DIR)/golangci-lint
+GOPATH           ?= $(shell go env GOPATH)
+BIN_DIR          := $(GOPATH)/bin
+GOLANGCI_LINT    := $(BIN_DIR)/golangci-lint
+
+ifeq ($(OS),Windows_NT)
+    OSName       ?= windows
+else
+    OSName       ?= $(shell echo $(shell uname -s) | tr '[:upper:]' '[:lower:]')
```

Comment:
"echo" is unnecessary.
`$(shell uname -s | tr '[:upper:]' '[:lower:]')` is ok.

Reply:
Yes, I will change this line like this:

``` bash
$(shell uname -s | tr '[:upper:]' '[:lower:]')
```